system_tests/bigquery.py

```diff
@@ -90,7 +92,15 @@ def test_update_dataset(self):
         after = [grant for grant in dataset.access_grants
                  if grant.entity_id != 'projectWriters']
         dataset.access_grants = after
-        dataset.update()
+
+        # We should try and keep `delay` and tries as low as possible to
+        # reduce test running time. tries=3 and delay=30 worked consistenly
+        # when updating a dataset.
```

Comment:
@jgeewax Can you help.

Returning to the original question, did you try less than 3 retries? Did you try less than 30s? What happened when you did?